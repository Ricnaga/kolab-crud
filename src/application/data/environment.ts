export const API_URL = String(import.meta.env.VITE_API_URL);
export const USER_ID = String(import.meta.env.VITE_USER_ID);
export const IS_DEVELOPMENT = !!import.meta.env.DEV;