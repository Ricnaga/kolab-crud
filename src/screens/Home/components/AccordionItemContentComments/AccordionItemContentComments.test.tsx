import { USER_ID } from "@application/data/environment";
import { Accordion, Card } from "@chakra-ui/react";
import {
  beforeEach,
  describe,
  expect,
  it,
  RenderResult,
  renderWithTheme,
  vi,
} from "@vitest-utils";
import { AccordionItemContentComments } from ".";

let layout: RenderResult;

describe("Home Component: AccordionItemContentComments", () => {
  beforeEach(() => {
    vi.resetModules();

    const screen = renderWithTheme(
      <Accordion.Root>
        <Accordion.Item value="0">
          <Card.Root>
            <AccordionItemContentComments
              isEnabled={true}
              post={{
                body: "BODY_MOCKED",
                id: 0,
                title: "title MOCKED",
                userId: parseInt(USER_ID, 10),
              }}
            />
          </Card.Root>
        </Accordion.Item>
      </Accordion.Root>
    );

    layout = screen;
  });

  it("should render correctly", () => {
    vi.doMock("@/infra/comments", () => ({
      useCommentsQuery: () => ({ data: [], isLoading: false }),
    }));

    vi.doMock("@/infra/users", () => ({
      useUserQuery: () => ({ data: null, isLoading: false }),
    }));

    expect(layout).toMatchSnapshot();
  });
  
});
