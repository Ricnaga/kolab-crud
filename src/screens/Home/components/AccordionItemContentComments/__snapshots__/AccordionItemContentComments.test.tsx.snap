// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component: AccordionItemContentComments > should render comments loading state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <div
        class="css-1f9j1y1"
      >
        <div
          class="chakra-accordion__root css-7u3wtz"
          data-orientation="vertical"
          data-part="root"
          data-scope="accordion"
          dir="ltr"
          id="accordion::r2:"
        >
          <div
            class="chakra-accordion__item css-xiajt7"
            data-orientation="vertical"
            data-part="item"
            data-scope="accordion"
            data-state="closed"
            dir="ltr"
            id="collapsible:accordion::r2::item:0"
          >
            <div
              class="chakra-card__root css-fgvmqn"
            >
              <div
                class="css-1tuimaw"
              >
                <div
                  class="chakra-card__root css-1kfp14p"
                >
                  <div
                    class="chakra-card__header css-aesgey"
                  >
                    <div
                      class="chakra-skeleton css-1eul3x0"
                    />
                  </div>
                  <div
                    class="chakra-card__body css-11e8ur2"
                  >
                    <div
                      class="chakra-skeleton css-7ftq5l"
                    />
                  </div>
                </div>
                <div
                  class="chakra-card__root css-1kfp14p"
                >
                  <div
                    class="chakra-card__header css-aesgey"
                  >
                    <div
                      class="chakra-skeleton css-1eul3x0"
                    />
                  </div>
                  <div
                    class="chakra-card__body css-11e8ur2"
                  >
                    <div
                      class="chakra-skeleton css-7ftq5l"
                    />
                  </div>
                </div>
                <div
                  class="chakra-card__root css-1kfp14p"
                >
                  <div
                    class="chakra-card__header css-aesgey"
                  >
                    <div
                      class="chakra-skeleton css-1eul3x0"
                    />
                  </div>
                  <div
                    class="chakra-card__body css-11e8ur2"
                  >
                    <div
                      class="chakra-skeleton css-7ftq5l"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1f9j1y1"
    >
      <div
        class="chakra-accordion__root css-7u3wtz"
        data-orientation="vertical"
        data-part="root"
        data-scope="accordion"
        dir="ltr"
        id="accordion::r2:"
      >
        <div
          class="chakra-accordion__item css-xiajt7"
          data-orientation="vertical"
          data-part="item"
          data-scope="accordion"
          data-state="closed"
          dir="ltr"
          id="collapsible:accordion::r2::item:0"
        >
          <div
            class="chakra-card__root css-fgvmqn"
          >
            <div
              class="css-1tuimaw"
            >
              <div
                class="chakra-card__root css-1kfp14p"
              >
                <div
                  class="chakra-card__header css-aesgey"
                >
                  <div
                    class="chakra-skeleton css-1eul3x0"
                  />
                </div>
                <div
                  class="chakra-card__body css-11e8ur2"
                >
                  <div
                    class="chakra-skeleton css-7ftq5l"
                  />
                </div>
              </div>
              <div
                class="chakra-card__root css-1kfp14p"
              >
                <div
                  class="chakra-card__header css-aesgey"
                >
                  <div
                    class="chakra-skeleton css-1eul3x0"
                  />
                </div>
                <div
                  class="chakra-card__body css-11e8ur2"
                >
                  <div
                    class="chakra-skeleton css-7ftq5l"
                  />
                </div>
              </div>
              <div
                class="chakra-card__root css-1kfp14p"
              >
                <div
                  class="chakra-card__header css-aesgey"
                >
                  <div
                    class="chakra-skeleton css-1eul3x0"
                  />
                </div>
                <div
                  class="chakra-card__body css-11e8ur2"
                >
                  <div
                    class="chakra-skeleton css-7ftq5l"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home Component: AccordionItemContentComments > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <div
        class="css-1f9j1y1"
      >
        <div
          class="chakra-accordion__root css-7u3wtz"
          data-orientation="vertical"
          data-part="root"
          data-scope="accordion"
          dir="ltr"
          id="accordion::r0:"
        >
          <div
            class="chakra-accordion__item css-xiajt7"
            data-orientation="vertical"
            data-part="item"
            data-scope="accordion"
            data-state="closed"
            dir="ltr"
            id="collapsible:accordion::r0::item:0"
          >
            <div
              class="chakra-card__root css-fgvmqn"
            >
              <div
                class="css-1tuimaw"
              >
                <div
                  class="chakra-card__root css-1kfp14p"
                >
                  <div
                    class="chakra-card__header css-aesgey"
                  >
                    <div
                      class="chakra-skeleton css-1eul3x0"
                    />
                  </div>
                  <div
                    class="chakra-card__body css-11e8ur2"
                  >
                    <div
                      class="chakra-skeleton css-7ftq5l"
                    />
                  </div>
                </div>
                <div
                  class="chakra-card__root css-1kfp14p"
                >
                  <div
                    class="chakra-card__header css-aesgey"
                  >
                    <div
                      class="chakra-skeleton css-1eul3x0"
                    />
                  </div>
                  <div
                    class="chakra-card__body css-11e8ur2"
                  >
                    <div
                      class="chakra-skeleton css-7ftq5l"
                    />
                  </div>
                </div>
                <div
                  class="chakra-card__root css-1kfp14p"
                >
                  <div
                    class="chakra-card__header css-aesgey"
                  >
                    <div
                      class="chakra-skeleton css-1eul3x0"
                    />
                  </div>
                  <div
                    class="chakra-card__body css-11e8ur2"
                  >
                    <div
                      class="chakra-skeleton css-7ftq5l"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1f9j1y1"
    >
      <div
        class="chakra-accordion__root css-7u3wtz"
        data-orientation="vertical"
        data-part="root"
        data-scope="accordion"
        dir="ltr"
        id="accordion::r0:"
      >
        <div
          class="chakra-accordion__item css-xiajt7"
          data-orientation="vertical"
          data-part="item"
          data-scope="accordion"
          data-state="closed"
          dir="ltr"
          id="collapsible:accordion::r0::item:0"
        >
          <div
            class="chakra-card__root css-fgvmqn"
          >
            <div
              class="css-1tuimaw"
            >
              <div
                class="chakra-card__root css-1kfp14p"
              >
                <div
                  class="chakra-card__header css-aesgey"
                >
                  <div
                    class="chakra-skeleton css-1eul3x0"
                  />
                </div>
                <div
                  class="chakra-card__body css-11e8ur2"
                >
                  <div
                    class="chakra-skeleton css-7ftq5l"
                  />
                </div>
              </div>
              <div
                class="chakra-card__root css-1kfp14p"
              >
                <div
                  class="chakra-card__header css-aesgey"
                >
                  <div
                    class="chakra-skeleton css-1eul3x0"
                  />
                </div>
                <div
                  class="chakra-card__body css-11e8ur2"
                >
                  <div
                    class="chakra-skeleton css-7ftq5l"
                  />
                </div>
              </div>
              <div
                class="chakra-card__root css-1kfp14p"
              >
                <div
                  class="chakra-card__header css-aesgey"
                >
                  <div
                    class="chakra-skeleton css-1eul3x0"
                  />
                </div>
                <div
                  class="chakra-card__body css-11e8ur2"
                >
                  <div
                    class="chakra-skeleton css-7ftq5l"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
